runtime/src/main/java/org/corfudb/runtime/collections/TransactionPoller.java

```diff
@@ -69,18 +83,12 @@ private void pollTxStream() {
  
         long lastReadAddress = streamContexts.get(0).getLastReadAddress();
 
-        StreamOptions options = StreamOptions.builder()
-                .ignoreTrimmed(true)
-                .cacheEntries(false)
-                .build();
-        IStreamView txStream = runtime.getStreamsView()
-                .get(ObjectsView.TRANSACTION_STREAM_ID, options);
         log.trace("Seeking txStream to {}", lastReadAddress + 1);
-        txStream.seek(lastReadAddress + 1);
+        txnStream.seek(lastReadAddress + 1);
```

Comment:
If a trim exception happens, then lastReadAddress won't be updated, no?

At some point you want to seek to the stream tail, so you need to query the sequencer and seek to that point.